api/ads.proto

```diff
@@ -0,0 +1,24 @@
+syntax = "proto3";
+
+package envoy.api.v2;
+
+import "api/base.proto";
+import "api/cds.proto";
+import "api/eds.proto";
+import "api/lds.proto";
+import "api/rds.proto";
+
+import "google/api/annotations.proto";
+
+// See https://github.com/lyft/envoy-api#apis for a description of the role of
+// ADS and how it is intended to be used by a management server. ADS requests
+// have the same structure as their singleton xDS counterparts, but can
+// multiplex many resource types on a single stream. The type_url in the
+// DiscoveryRequest/DiscoveryResponse provides sufficient information to recover
+// the multiplexed singleton APIs at the Envoy instance and management server.
+service AggregatedDiscoveryService {
+  // This is a gRPC-only API.
+  rpc StreamAggregatedResources(stream DiscoveryRequest)
+      returns (stream DiscoveryResponse) {
```

Comment:
Even though we have dropped the canary conversion for now, do we still want to make the API a stream of "atomic" lists of xDS updates? I see now downside since from an implementation perspective initially it could just still go through them in order, but in the future it would allow us to to "atomic" things more easily without an API change?